pkg/volume/csi/csi_mounter.go

```diff
@@ -187,9 +238,7 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 		return err
 	}
 
-	err := csi.NodeUnpublishVolume(ctx, volID, dir)
-
-	if err != nil {
+	if err := csi.NodeUnpublishVolume(ctx, volID, dir); err != nil {
 		glog.Errorf(log("Mounter.Setup failed: %v", err))
 		return err
 	}
```

Comment:
We should make sure to delete the `volData` and delete the volume directory (that we created). But we have to be careful deleting the directory, we should test if it is a mount point or not. Basically do what `UnmountMountPoint(...)` does in https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go#L98 but don't unmount, and just fail if not unmounted.

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go#L98

Reply:
@saad-ali when I read this couple days ago I did not understand a word.... man, the power of rest.  Ok, will take care of this.